x-pack/packages/ai-infra/kb-artifact-builder/src/build_artifacts.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Path from 'path';
+import { Client } from '@elastic/elasticsearch';
+import { ToolingLog } from '@kbn/tooling-log';
+import {
+  // checkConnectivity,
+  createTargetIndex,
+  extractDocumentation,
+  indexDocuments,
+  installElser,
+  createChunkFiles,
+  createArtifact,
+  cleanupFolders,
+} from './tasks';
+import type { TaskConfig } from './types';
+
+const getSourceClient = (config: TaskConfig) => {
+  return new Client({
+    compression: true,
+    nodes: [config.sourceClusterUrl],
+    sniffOnStart: false,
+    auth: {
+      username: config.sourceClusterUsername,
+      password: config.sourceClusterPassword,
+    },
+  });
+};
+
+const getEmbeddingClient = (config: TaskConfig) => {
+  return new Client({
+    compression: true,
+    nodes: [config.embeddingClusterUrl],
+    auth: {
+      username: config.embeddingClusterUsername,
+      password: config.embeddingClusterPassword,
+    },
+    // generating embeddings takes time
+    requestTimeout: 10 * 60 * 1000,
+  });
+};
+
+export const buildArtifacts = async (config: TaskConfig) => {
+  const log = new ToolingLog({
+    level: 'info',
+    writeTo: process.stdout,
+  });
+
+  log.info(
+    `Starting building artifacts for version=${
+      config.stackVersion
+    } and products=[${config.productNames.join(',')}]`
+  );
+
+  const sourceClient = getSourceClient(config);
+  const embeddingClient = getEmbeddingClient(config);
+
+  // log.info('Checking connectivity against clusters');
+  // await checkConnectivity({ sourceClient, embeddingClient });
```

Comment:
Don't have ping permission on the source cluster, had to disable that for now...